apps/src/lib/levelbuilder/lesson-editor/ObjectivesEditor.jsx

```diff
@@ -0,0 +1,209 @@
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+import color from '@cdo/apps/util/color';
+
+const styles = {
+  oddRow: {
+    backgroundColor: color.lightest_gray
+  },
+  actionButtons: {
+    display: 'flex',
+    justifyContent: 'space-evenly',
+    backgroundColor: 'white',
+    height: 30
+  },
+  remove: {
+    fontSize: 14,
+    color: 'white',
+    background: color.dark_red,
+    cursor: 'pointer',
+    textAlign: 'center',
+    minWidth: '50%',
+    lineHeight: '30px'
+  },
+  edit: {
+    fontSize: 14,
+    color: 'white',
+    background: color.default_blue,
+    cursor: 'pointer',
+    textAlign: 'center',
+    minWidth: '50%',
+    lineHeight: '30px'
+  },
+  save: {
+    fontSize: 14,
+    color: 'white',
+    background: color.green,
+    cursor: 'pointer',
+    textAlign: 'center',
+    minWidth: '50%',
+    lineHeight: '30px'
+  },
+  addButton: {
+    background: color.cyan,
+    borderRadius: 3,
+    color: color.white,
+    fontSize: 14,
+    padding: 7,
+    textAlign: 'center',
+    marginTop: 10,
+    marginLeft: 0
+  }
+};
+
+export default class ObjectivesEditor extends Component {
+  static propTypes = {
+    objectives: PropTypes.arrayOf(PropTypes.object)
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      objectives: props.objectives || [],
+      objectiveInput: '',
+      currentlyEditingIndex: null,
+      nextObjectiveKey: 1
+    };
+  }
+
+  handleRemove = idx => {
+    let {objectives} = this.state;
+    objectives.splice(idx, 1);
+    this.setState({objectives});
+  };
+
+  handleEdit = idx => {
+    this.setState({
+      currentlyEditingIndex: idx,
+      objectiveInput: this.state.objectives[idx].description
+    });
+  };
+
+  handleCancel = () => {
+    let {objectives, objectiveInput, currentlyEditingIndex} = this.state;
+    if (
+      objectiveInput === '' &&
+      currentlyEditingIndex === objectives.length - 1
+    ) {
+      objectives.splice(objectives.length - 1, 1);
```

Comment:
I think you need to make a copy again here for objectives since you are modifying it